packages/block-library/src/navigation/edit/index.js

```diff
@@ -257,6 +270,17 @@ function Navigation( {
 		hasResolvedNavigationMenus &&
 		! hasUncontrolledInnerBlocks;
 
+	if ( isPlaceholder && ! ref ) {
+		/**
+		 *  this fallback only displays (both in editor and on front)
+		 *  the list of pages block if not menu is available
+		 *  we don't want the fallback to request a save
+		 *  nor to be undoable, hence we mark it non persistent
+		 */
+		__unstableMarkNextChangeAsNotPersistent();
+		replaceInnerBlocks( clientId, [ createBlock( 'core/page-list' ) ] );
```

Comment:
It looks like this action is dispatched on render, so have to be a little bit careful about accidentally dispatching it multiple times. 

Using the inner blocks `template` property may also be a viable way to define some default blocks. Though I might be missing some aspects of the requirements here.

Reply:
@talldan is [this](https://github.com/WordPress/gutenberg/pull/42735#discussion_r932981300) a simialar idea?